### ISSUE TITLE
Unable to create SASsession in MacOS 12 (Monterey) beta

### ISSUE THREAD

**Issue by arge-7** [sassoftware/saspy#395]
**Describe the bug**

Unable to create a new session via saspy.SASsession() on M1 Mac mini running the MacOS 12 beta (Monterey). It works fine on my Intel Mac running OS 11, but I'm unable to create a session in a Jupyter notebook. I don't know if this is an issue on Apple's end or yours, but it seems to be related to issues with Java. Saspy was definitely working on OS 11, because I was using it frequently on this device but was curious to try the beta of the next major MacOS.

**To Reproduce**

1. Open Jupyter notebook via Jupyter Lab on M1 Mac running MacOS 12
2. import saspy
3. Try to create a session via `sas_session=saspy.SASsession()`

**Expected behavior**

sas session is created and I can use saspy as usual.

**Desktop (please complete the following information):**
 - OS: MacOS 12 beta
 - SAS Version: onDemand for academics
 - SASPy Version: 3.7.2

**Additional context**

This is the error generated immediately after trying to create the session in a Jupyter notebook (but the same thing happens when running Python via terminal)

```

Using SAS Config named: oda

SAS process has terminated unexpectedly. Pid State= (14710, 6)
SAS Connection failed. No connection established. Double check your settings in sascfg_personal.py file.

Attempted to run program /usr/bin/java with the following parameters: ['/usr/bin/java', '-classpath',  ...

```
followed by many .jar file paths. and the error `No SAS process attached. SAS process has terminated unexpectedly.
Invalid response from SAS on inital submission. printing the SASLOG as diagnostic`

I don't know which parts of the long log file generated are useful, but this seemed to be the most relevant:

```

Internal exceptions (18 events):
Event: 0.178 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfcc0fd0}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.lang.Object)'> (0x00000006cfcc0fd0) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.179 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfcc4720}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.invokeStaticInit(java.lang.Object)'> (0x00000006cfcc4720) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.179 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfccd568}: 'java.lang.Object java.lang.invoke.Invokers$Holder.invokeExact_MT(java.lang.Object, java.lang.Object, java.lang.Object, java.lang.Object, java.lang.Object, java.lang.Object, java.lang.Object, java.lang.Object)'> (0x00000006cfccd568) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.182 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfcd8a60}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.lang.Object, java.lang.Object, java.lang.Object)'> (0x00000006cfcd8a60) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.182 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfce1268}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.getReference(java.lang.Object, java.lang.Object)'> (0x00000006cfce1268) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.182 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfce5620}: 'java.lang.Object java.lang.invoke.LambdaForm$Holder.identity_L(java.lang.Object, java.lang.Object)'> (0x00000006cfce5620) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.183 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfce8560}: 'java.lang.Object java.lang.invoke.LambdaForm$Holder.zero_L(java.lang.Object)'> (0x00000006cfce8560) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.183 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfceec48}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.invokeSpecial(java.lang.Object, java.lang.Object)'> (0x00000006cfceec48) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.183 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfcf2280}: 'java.lang.Object java.lang.invoke.Invokers$Holder.linkToTargetMethod(java.lang.Object)'> (0x00000006cfcf2280) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.197 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfcfe580}: 'void java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.lang.Object, java.lang.Object)'> (0x00000006cfcfe580) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.198 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfd043e8}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.newInvokeSpecial(java.lang.Object, java.lang.Object)'> (0x00000006cfd043e8) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.198 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfd08740}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.invokeSpecial(java.lang.Object, java.lang.Object, java.lang.Object)'> (0x00000006cfd08740) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.198 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfd0bc28}: 'java.lang.Object java.lang.invoke.Invokers$Holder.linkToTargetMethod(java.lang.Object, java.lang.Object)'> (0x00000006cfd0bc28) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.826 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfd80840}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.lang.Object, java.lang.Object)'> (0x00000006cfd80840) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.840 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfa029a8}: 'int java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.lang.Object, int)'> (0x00000006cfa029a8) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.841 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfa08ea0}: 'int java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(java.lang.Object, int, int)'> (0x00000006cfa08ea0) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.841 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfa0ee48}: 'java.lang.Object java.lang.invoke.DirectMethodHandle$Holder.newInvokeSpecial(java.lang.Object, int)'> (0x00000006cfa0ee48) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]
Event: 0.841 Thread 0x000000015e00d400 Exception <a 'java/lang/NoSuchMethodError'{0x00000006cfa162d0}: 'java.lang.Object java.lang.invoke.Invokers$Holder.linkToTargetMethod(int, java.lang.Object)'> (0x00000006cfa162d0) 
thrown [src/hotspot/share/interpreter/linkResolver.cpp, line 790]

```

I figured I would ask since this could be a problem once OS 12 is publicly released. I'm sure there's more info that would be helpful but I didn't want to include endless lines of logs... Thanks!

**Comment by tomweber-sas** [comment]
Well, I don't see anything in that that clues me in, other than some java problem. How about email me the entire thing, and maybe I'll see something more that'll help. Do you know what version of java this is? And can you send your config info; your sascfg_personal.py? 

Thanks!
Tom

**Comment by tomweber-sas** [comment]
I replied in email, but figured I'd add it here incase you didn't get it.
```
Well, the sascfg_personal.py didn’t come through, since it has the .py extension; security. If you
just add .txt to it, it’ll come through. I’ve looked at the log and the traceback shows that it’s trying
to make the connection to SAS. It’s not clear to me why it’s getting an exception. Is there something
on the system blocking it from being able to make a connection? I don’t have a way to try to reproduce
this. I do wonder about the java version, as that’s the only other obvious thing. It’s failing in java trying
to connect. Does Java 8 that you’re installing behave any differently?

Thanks,
Tom
```
Did you install java 8, and did it make a difference? FYI, the traceback is that first section of the log file you sent. And, again, it is the call to make the connection to SAS that gets a java exception, which I've never see before, and isn't clear as to what the issue is. 

Thanks,
Tom

**Comment by tomweber-sas** [comment]
It seems you're java is java 16. So, I installed that on my windows box and tried connecting to ODA with that, in case there is some compatibility issue w/ our java IOM client (the jars in saspy). It worked without a problem, so at least that's a good sign, though it doesn't help me see what could be the problem on your end. 
Guess I'd still like to see your config file, for the sake of that.
```
C:\Users\sastpw>"C:\Program Files\Java\jdk-16.0.2\bin\java.exe" --version
java 16.0.2 2021-07-20
Java(TM) SE Runtime Environment (build 16.0.2+7-67)
Java HotSpot(TM) 64-Bit Server VM (build 16.0.2+7-67, mixed mode, sharing)



sasc = saspy.SASsession(cfgname='oda', java=r'C:\Program Files\Java\jdk-16.0.2\bin\java.exe')
sasc

SAS Connection established. Subprocess id is 31208

No encoding value provided. Will try to determine the correct encoding.
Setting encoding to utf_8 based upon the SAS session encoding value of utf-8.

Access Method         = IOM
SAS Config name       = oda
SAS Config file       = C:\ProgramData\Anaconda3\lib\site-packages\saspy\sascfg_personal.py
WORK Path             = /saswork/SAS_workC2BF00018A4A_odaws02-usw2.oda.sas.com/SAS_work541600018A4A_odaws02-usw2.oda.sas.com/
SAS Version           = 9.04.01M6P11072018
SASPy Version         = 3.7.2
Teach me SAS          = False
Batch                 = False
Results               = Pandas
SAS Session Encoding  = utf-8
Python Encoding value = utf_8
SAS process Pid value = 100938
```

**Comment by arge-7** [comment]
So sorry for the delay. Yes I got your email. I went to check my Java version in the terminal, and actually got a Java error just trying to do that. I searched for the error on Stackoverflow, and found another report of it with a fix (https://stackoverflow.com/questions/67684430/apple-siliconm1-11-4-get-a-java-error-jdk16 if you're curious). I followed the advice to use 'Zulu' which fixed it instantly after installing it.

Thanks for your help!
